src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/services/AuditS3Client.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.digital.hmpps.hmppsauditapi.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.core.sync.RequestBody
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.PutObjectRequest
+import uk.gov.justice.digital.hmpps.hmppsauditapi.listeners.HMPPSAuditListener
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+import java.time.ZoneId
+
+@Service
+class AuditS3Client(
+  private val s3Client: S3Client,
+  private val objectMapper: ObjectMapper,
+  @Value("\${aws.s3.auditBucketName}") private val bucketName: String,
+) {
+
+  fun save(auditEvent: HMPPSAuditListener.AuditEvent) {
```

Comment:
What happens in the event of a failure? (temporary, maybe can't connect to the S3 API, in which case we can retry) or permanent (eg there's no when or who fields in the user event?)

Reply:
at the minute the request will just fall over. What's our retry policy? re body validation- that happens well before we get to this point so we don't need to worry about audit events have dirty data